### PR TITLE
ARROW-3062: [Python] Fix python package finder to also work in Python 2.7

### PR DESCRIPTION
**Pull request by pcmoritz**

### CODE REVIEW

**Review by pitrou — APPROVED** [review]
All this is ugly, but I guess it's ok...

File: python/pyarrow/compat.py [hunk 1]
```diff
@@ -164,34 +164,43 @@ def _iterate_python_module_paths(package_name):
...
-
-    if spec:
-        module = importlib.util.module_from_spec(spec)
-        for path in module.__path__:
-            yield path
+        try:
+            import importlib
+            absolute_name = importlib.util.resolve_name(package_name, None)
+        except (ImportError, AttributeError):
+            # Sometimes, importlib is not available (e.g. Python 2)
```
Comment (cpcloud) [inline python/pyarrow/compat.py]: We're not supporting anything older than python 2.7 right? importlib was added in 2.7 so this should never throw an import error.

Comment (wesm) [reply python/pyarrow/compat.py]: Right, nothing older than 2.7

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**